packages/node-core/src/configure/migration-service/migration.ts

```diff
@@ -93,46 +191,80 @@ export class Migration {
       this.sequelizeModels.push(sequelizeModel);
     }
   }
 
-  private createModel(model: GraphQLModelsType) {
+  private createSequelizeModel(model: GraphQLModelsType): ModelStatic<any> {
     const {attributes, indexes} = this.prepareModelAttributesAndIndexes(model);
     return this.storeService.defineModel(model, attributes, indexes, this.schemaName);
   }
 
-  async createTable(model: GraphQLModelsType): Promise<void> {
+  async createTable(model: GraphQLModelsType, withoutForeignKey: boolean): Promise<void> {
     const {attributes, indexes} = this.prepareModelAttributesAndIndexes(model);
+    const [indexesResult] = await this.sequelize.query(getExistedIndexesQuery(this.schemaName));
+    const existedIndexes = indexesResult.map((i) => (i as any).indexname);
 
     if (indexes.length > this.config.indexCountLimit) {
       throw new Error(`too many indexes on entity ${model.name}`);
     }
 
-    const [indexesResult] = await this.sequelize.query(getExistedIndexesQuery(this.schemaName));
-    const existedIndexes = indexesResult.map((i) => (i as any).indexname);
-
     if (this.historical) {
       addBlockRangeColumnToIndexes(indexes);
       addHistoricalIdIndex(model, indexes);
     }
 
-    updateIndexesName(model.name, indexes, existedIndexes);
+    updateIndexesName(model.name, indexes, existedIndexes as string[]);
+    // Update index query for cockroach db
+    this.beforeHandleCockroachIndex(this.schemaName, model.name, indexes, existedIndexes as string[]);
 
     const sequelizeModel = this.storeService.defineModel(model, attributes, indexes, this.schemaName);
 
-    this.addModel(sequelizeModel);
+    this.mainQueries.push(...generateCreateTableStatement(sequelizeModel, this.schemaName, withoutForeignKey));
+
+    if (sequelizeModel.options.indexes) {
+      this.mainQueries.push(
+        ...generateCreateIndexStatement(sequelizeModel.options.indexes, this.schemaName, sequelizeModel.tableName)
+      );
+    }
+
+    if (this.useSubscription) {
+      const triggerName = hashName(this.schemaName, 'notify_trigger', sequelizeModel.tableName);
+      const notifyTriggers = await getTriggers(this.sequelize, triggerName);
+      // Triggers not been found
+      if (notifyTriggers.length === 0) {
+        this.extraQueries.push(createNotifyTrigger(this.schemaName, sequelizeModel.tableName));
+      } else {
+        validateNotifyTriggers(triggerName, notifyTriggers as NotifyTriggerPayload[]);
+      }
+    } else {
+      //TODO: DROP TRIGGER IF EXIST is not valid syntax for cockroach, better check trigger exist at first.
+      if (this.dbType !== SUPPORT_DB.cockRoach) {
+        this.extraQueries.push(dropNotifyTrigger(this.schemaName, sequelizeModel.tableName));
+      }
+    }
+
+    if (!this.useSubscription && this.dbType !== SUPPORT_DB.cockRoach) {
+      this.extraQueries.push(dropNotifyFunction(this.schemaName));
+    }
+
+    this.addModelToSequelizeCache(sequelizeModel);
```

Comment:
Previously this would have been run every startup, now it will only be run on migrations